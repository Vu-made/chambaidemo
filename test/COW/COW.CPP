#include <bits/stdc++.h>
#define pii pair<int,int>
#define ll long long
#define N 100005
using namespace std;
pii a[N];
int n,d;
ll Smax = 0;
int main()
{
    freopen("COW.INP","r",stdin);
    freopen("COW.OUT","w",stdout);
    cin >> n >> d;
    for (int i=1;i<=n; i++) cin >> a[i].first >> a[i].second;
    sort(a+1,a+n+1);
    int i = 1, j = 2;
    ll S = Smax = a[1].second;
    while (i <= j && j <= n)
    {
        if (a[j].first - a[i].first <= d)
        {
            S += a[j++].second;
            Smax = max(Smax,S);
        }
        else S -= a[i++].second;
    }
    cout << Smax;
    return 0;
}
