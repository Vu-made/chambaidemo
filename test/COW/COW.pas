const   fi = 'COW.INP';
        fo = 'COW.OUT';
        nmax = 100001;
var     a,b:array[1..nmax] of longint;
        n,d,Smax:longint;
procedure nhap;
var     f:text;
        i:longint;
begin
        assign(f,fi); reset(f);
        read(f,n,d);
        for i:=1 to n do readln(f,a[i],b[i]);
        close(f);
end;

procedure hoandoi(var x,y:longint);
var     tg:longint;
begin
        tg := x; x := y; y:=tg;
end;

procedure QS(d,c:longint);
var     i,j,chot:longint;
begin
        i := d; j := c; chot := a[(i+j) div 2];
        repeat
                while (a[i] < chot) do inc(i);
                while (chot < a[j]) do dec(j);
                if (i <= j) then
                begin
                        hoandoi(a[i],a[j]);
                        hoandoi(b[i],b[j]);
                        inc(i); dec(j);
                end;
        until i > j;
        if (i < c) then QS(i,c);
        if (d < j) then QS(d,j);
end;


procedure xuly;
var     S,i,j:longint;
begin
        QS(1,n);
        i := 1; j:=2;
        S := b[1]; Smax := b[1];
        while (i <= j) and (j <= n) do
        begin
                if (a[j] - a[i] <= d) then
                begin
                        S := S + b[j];
                        inc(j);
                        if (Smax < S) then Smax := S;
                end
                else    begin
                                S := S - b[i];
                                inc(i);
                        end;
        end;
end;

procedure ghi;
var     f:text;
begin
        assign(f,fo); rewrite(f);
        write(f,Smax);
        close(f);
end;

Begin
        nhap;
        xuly;
        ghi;
end.