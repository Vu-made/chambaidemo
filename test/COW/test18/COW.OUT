41608412495654